src/sign.c

```diff
@@ -22,88 +22,87 @@ typedef struct sign sign_T;
+    int sn_num_hl;   // highlight ID for line number
+    int sn_priority; // default priority of this sign, -1 means SIGN_DEF_PRIO
 };
 
-static sign_T	*first_sign = NULL;
-static int	next_sign_typenr = 1;
+static sign_T *first_sign = NULL;
+static int next_sign_typenr = 1;
 
 static void sign_list_defined(sign_T *sp);
 static void sign_undefine(sign_T *sp, sign_T *sp_prev);
 
-static char *cmds[] = {
-			"define",
-# define SIGNCMD_DEFINE	0
-			"undefine",
-# define SIGNCMD_UNDEFINE 1
-			"list",
-# define SIGNCMD_LIST	2
-			"place",
-# define SIGNCMD_PLACE	3
-			"unplace",
-# define SIGNCMD_UNPLACE 4
-			"jump",
-# define SIGNCMD_JUMP	5
-			NULL
-# define SIGNCMD_LAST	6
+static char *cmds[] = { "define",
+#define SIGNCMD_DEFINE 0
+                        "undefine",
+#define SIGNCMD_UNDEFINE 1
+                        "list",
+#define SIGNCMD_LIST 2
+                        "place",
+#define SIGNCMD_PLACE 3
+                        "unplace",
+#define SIGNCMD_UNPLACE 4
+                        "jump",
+#define SIGNCMD_JUMP 5
+                        NULL
+#define SIGNCMD_LAST 6
 };
 
-#define FOR_ALL_SIGNS(sp)	\
+#define FOR_ALL_SIGNS(sp) \
     for ((sp) = first_sign; (sp) != NULL; (sp) = (sp)->sn_next)
 
-static hashtab_T	sg_table;	// sign group (signgroup_T) hashtable
-static int		next_sign_id = 1; // next sign id in the global group
+static hashtab_T sg_table;   // sign group (signgroup_T) hashtable
+static int next_sign_id = 1; // next sign id in the global group
 
-/*
+                             /*
  * Initialize data needed for managing signs
  */
-    void
+void
 init_signs(void)
 {
-    hash_init(&sg_table);		// sign group hash table
+    hash_init(&sg_table); // sign group hash table
 }
 
 /*
  * A new sign in group 'groupname' is added. If the group is not present,
  * create it. Otherwise reference the group.
  */
-    static signgroup_T *
+static signgroup_T *
 sign_group_ref(char_u *groupname)
 {
-    hash_T		hash;
-    hashitem_T		*hi;
-    signgroup_T		*group;
+    hash_T hash;
+    hashitem_T *hi;
+    signgroup_T *group;
 
     hash = hash_hash(groupname);
     hi = hash_lookup(&sg_table, groupname, hash);
     if (HASHITEM_EMPTY(hi))
     {
-	// new group
-	group = alloc(offsetof(signgroup_T, sg_name) + STRLEN(groupname) + 1);
-	if (group == NULL)
-	    return NULL;
-	STRCPY(group->sg_name, groupname);
-	group->sg_refcount = 1;
-	group->sg_next_sign_id = 1;
-	hash_add_item(&sg_table, hi, group->sg_name, hash);
+        // new group
+        group = alloc(offsetof(signgroup_T, sg_name) + STRLEN(groupname) + 1);
+        if (group == NULL)
+            return NULL;
+        STRCPY(group->sg_name, groupname);
+        group->sg_refcount = 1;
+        group->sg_next_sign_id = 1;
+        hash_add_item(&sg_table, hi, group->sg_name, hash);
     }
     else
     {
-	// existing group
-	group = HI2SG(hi);
-	group->sg_refcount++;
+        // existing group
+        group = HI2SG(hi);
+        group->sg_refcount++;
```

Comment:
I don't think this PR should be merged.  This doesn't follow the existing coding convention (that Bram diligently maintained for 30 years).

Reply:
> This doesn't follow the existing coding convention (that Bram diligently maintained for 30 years)

The 'existing' coding convention is really lose and it isn't followed everywhere. It is mixed and matched and really hard to follow. A formatter with a set of rules can help with unifying the style